app/services/foreman/access_permissions.rb

```diff
@@ -19,7 +19,8 @@
     map.permission :edit_architectures,
                    :architectures => [:edit, :update],
                    :"api/v1/architectures" => [:update],
-                   :"api/v2/architectures" => [:update]
+                   :"api/v2/architectures" => [:update],
+                   :"api/v2/many_to_many" => [:create, :destroy]
```

Comment:
@isratrade, what about this suggestion:
I was thinking about the permissions model in the system, and I think that with a little effort in extending the permissions module, we can create a cleaner solution.
In the current code the smallest granule for permissions is `#controller/#action` pair. I suggest using the request URL and HTTP verb as a permissions granule, and use a "shortest match" search for each request to find the appropriate permission. I'll show you what I mean:
Let's say I want to GET /architectures/5/links/operatingsystems
the system will try to see if my user has a valid permission like this:
if I have permission to GET, /achitectures/* => return allow, else:
if I have permission to GET, /achitectures/*/links/* => return allow, else:
if I have permission to GET, /achitectures/*/links/operatingsystems => return allow, else: deny.

@ares, You are way more familiar with the authorization mechanism in the system, what do you think? Is it doable? Am I on the right track?